com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelController.java

```diff
@@ -139,16 +146,57 @@ public void update(Context ctx, String modeluri) {
 				handleError(ctx, 400, "Empty JSON");
 				return Optional.empty();
 			}
-			return codecs.decode(ctx, jsonData);
+			return codecs.decode(ctx, modelRepository.getResourceSet(), modelURI, jsonData);
 		} catch (DecodingException | IOException e) {
 			handleError(ctx, 400, "Invalid JSON");
 		}
 		return Optional.empty();
 	}
 
+	private  Optional<EObject> getContents(Optional<Resource> resource) {
+		return resource.map(r -> r.getContents().isEmpty() ? null : r.getContents().get(0));
```

Comment:
It seems as if we could end up with `null` wrapped in an `Optional` here.

Reply:
That's normal: the result will be an empty optional.  I'll add a doc comment.

Reply:
Cool, I didn't know that.